src/SonarScanner.MSBuild.PreProcessor/PreProcessor.cs

```diff
@@ -191,6 +192,21 @@ private async Task<ArgumentsAndRuleSets> FetchArgumentsAndRuleSets(ISonarWebServ
             return argumentsAndRuleSets;
         }
 
+        /// <summary>
+        /// We want the resource cache to be in a well-known location so we can re-use files that have
+        /// already been installed.
+        /// </summary>
+        private static string GetLocalCacheDirectory(string localCacheTempPath)
+        {
+            if (string.IsNullOrWhiteSpace(localCacheTempPath))
+            {
+                // here we will allow the Embedded Analyzer Installer to build out the temporary directory
+                return null;
+            }
+
+            return Path.Combine(localCacheTempPath, ".sonarqube", "resources");
+        }
+
```

Comment:
This can be removed and embedded into the `GetSetting` call as the default value